### DISCUSSION TITLE
Using Interactions with Storyshots (Property 'toBeVisible' does not exist on type 'JestMatchers<HTMLElement>')

### DISCUSSION THREAD

**Discussion by sophianathanielle** [storybookjs/storybook#17694] [Help] [answered]
Hello!
While using Interactions, I keep running into this error: `Property 'toBeVisible' does not exist on type 'JestMatchers<HTMLElement>'.`  and similar, coming from the `play` functions.
To get Storybook to open, I have added the following file to the `files` property in `.storybook/tsconfig.json`.
```
// setup.ts
import '@testing-library/jest-dom';
```

~~To get the Jest tests to run, I have added the same import to the `setup.ts` given in `setupFilesAfterEnv`.~~ This actually didn't work, never mind.

I am stumped in getting Interactions to work with ~~Storyshots~~ Jest in general. Did I go about this the wrong way? Any advice?

Storyshots file:
```
import initStoryshots, { multiSnapshotWithOptions, } from '@storybook/addon-storyshots';

initStoryshots({
  test: multiSnapshotWithOptions({}),
});
```

`package.json`:
```   
    "@storybook/addon-a11y": "^6.4.19",
    "@storybook/addon-actions": "^6.3.12",
    "@storybook/addon-essentials": "^6.3.12",
    "@storybook/addon-interactions": "^6.4.19",
    "@storybook/addon-links": "^6.3.12",
    "@storybook/addon-storyshots": "^6.4.19",
    "@storybook/angular": "^6.3.12",
    "@storybook/builder-webpack5": "^6.3.12",
    "@storybook/jest": "0.0.9",
    "@storybook/manager-webpack5": "^6.3.12",
    "@storybook/test-runner": "0.0.4",
    "@storybook/testing-library": "0.0.9",

    "@babel/preset-typescript": "^7.16.7",
    "@types/jest": "^27.4.0",
    "jest": "^27.5.1",
    "jest-preset-angular": "^11.1.1",
```

**Comment by sophianathanielle** [comment, marked as answer]
Hello!

Turns out I was importing the wrong thing. Adding `import '@testing-library/jest-dom/extend-expect';` to `setup.ts` (in Jest's `setupFilesAfterEnv`) fixed this for me. :)

**Comment by Kokolokoli** [comment]
A little late, but since Febuary the jest & jest-dom matchers are included automatically in `@storybook/jest` and available through the exposed `expect`. In other words, it should work out of the box, even without explicitly downloading the jest package.

See https://github.com/storybookjs/storybook/discussions/16858

Do note that in interactionsDebugger, all of the methods return promises that must be awaited. Not doing so may result in a similar error.
